driver-core/src/main/com/mongodb/internal/connection/InternalConnection.java

```diff
@@ -34,6 +35,13 @@
      */
     ConnectionDescription getDescription();
 
+    /**
+     * Get the initial server description
+     *
+     * @return the initial server description
+     */
+    ServerDescription getInitialServerDescription();
```

Comment:
This lets us support the requirement to update the cluster description based in the ismaster response from the initial handshake